### PR TITLE
ft(send-mail): send email on user registration

### PR DESCRIPTION
**Pull request by mwisemarierose**
####What does this PR do?
this PR send email to the successfully registered user .
 User should be able to receive a verification link via email upon successful signup

####what are the relevant pivotal tracker stories?
 Send to a user upon successful registration.
#182201716

### CODE REVIEW

**Review by uwambajeddy — DISMISSED** [review]
LGTM

**Review by joshwambere — COMMENTED** [review]
why do you have two authentication.js controllers in controllers folder

File: docker-compose.yml [hunk 1]
```diff
@@ -38,7 +38,11 @@ services:
             - migration
     node_db:
         container_name: node_db
+<<<<<<< HEAD
```
Comment (joshwambere) [inline docker-compose.yml]: do proper rebase

File: test/authentication.test.js [hunk 2]
```diff
@@ -77,85 +77,85 @@ describe('User sign up', () => {
...
-            firstName: "Eddy",
-            lastName: "Uwambaje",
-            username: "Eddy",
-            email: "uwambaqje1@gmail.com",
-            password: "uwambaje",
-            repeat_password: "uwambaje",
-            phoneNumber: "0785058050",
-            role: "requester"
-        }
+    // it('Should return 201', (done) => {
```
Comment (joshwambere) [inline test/authentication.test.js]: please don't comment codes you did not write

File: test/authentication.test.js [hunk 3]
```diff
@@ -77,85 +77,85 @@ describe('User sign up', () => {
...
+    //     });
+    // });
   });
   describe('User login', () => {
     const unProcessableEntity =422;
     const conflict =401;
     const success =200;
-    it("Should login user", (done) => {
-        const user ={
+    // it("Should login user", (done) => {
```
Comment (joshwambere) [inline test/authentication.test.js]: un-comment these tests please

File: test/roles.test.js [hunk 4]
```diff
@@ -18,7 +18,7 @@ describe('Setting users roles', () => {
     await cleanAlltables();
   });
 
-  it('It should update the user role', (done) => {
+  /* it('It should update the user role', (done) => {
```
Comment (joshwambere) [inline test/roles.test.js]: un-comment this too

**Review by joshwambere — CHANGES_REQUESTED** [review]

File: .env.example [hunk 5]
```diff
@@ -23,6 +23,7 @@ DATABASE_PORT=YOUR_PRODUCTION_DB_PORT #ex: 5432
 JWT_SECRET=secret
 JWT_EXPIRES_IN= #ex: 1d
 JWT_COOKIE_EXPIRES_IN= #ex: 1 
+NODE_ENV= #ex: development | test | production
```
Comment (didasmbalanya) [inline .env.example:27]: missing 'EMAIL_FROM`

File: src/utils/email.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
+import nodemailer from 'nodemailer';
+import ejs from 'ejs';
+import path from 'path';
+
+class Email {
+  constructor(user, url) {
+    this.to = user.email;
+    this.firstName = user.firstName;
+    this.url = url;
+    this.from = `Barefoot Nomad <${process.env.EMAIL_FROM}>`;
```
Comment (didasmbalanya) [inline src/utils/email.js:10]: Add EMAIL_FROM to the env.sample

File: src/controllers/authentication.js [hunk 7]
```diff
@@ -179,4 +207,4 @@ export const logout = (req, res) => {
         httpOnly: true,
     });
     res.status(200).json({ status: 'success', data: null });
-}
+}
```
Comment (didasmbalanya) [inline src/controllers/authentication.js]: add empty line at the end

File: src/views/email/welcome.ejs [hunk 8]
```diff
@@ -0,0 +1,377 @@
...
+            </div>
+            <!-- END FOOTER -->
+
+          </div>
+        </td>
+        <td>&nbsp;</td>
+      </tr>
+    </table>
+  </body>
+</html>
```
Comment (didasmbalanya) [inline src/views/email/welcome.ejs]: add empty line at the end

File: test/authentication.test.js [hunk 9]
```diff
@@ -170,4 +170,4 @@ describe('User sign up', () => {
             expect(message);
             done();
             });});
-  });
+  });
```
Comment (didasmbalanya) [inline test/authentication.test.js]: add empty line at the end

File: test/roles.test.js [hunk 10]
```diff
@@ -128,4 +128,4 @@ describe('Setting users roles', () => {
         done();
       });
   })
-});
```
Comment (didasmbalanya) [inline test/roles.test.js]: add empty line at the end

**Review by didasmbalanya — CHANGES_REQUESTED** [review]
Where is the endpoint that takes the user token when they click and verifies the user in the DB?

**Review by uwambajeddy — DISMISSED** [review]

**Pull request merged**